ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/AbstractBlockCreator.java

```diff
@@ -252,21 +249,20 @@ private BlockTransactionSelector.TransactionSelectionResults selectTransactions(
 
   private MutableWorldState duplicateWorldStateAtParent() {
     final Hash parentStateRoot = parentHeader.getStateRoot();
-    final MutableWorldState worldState =
-        protocolContext
-            .getWorldStateArchive()
-            .getMutable(parentStateRoot, parentHeader.getHash(), false)
-            .orElseThrow(
-                () -> {
-                  LOG.info("Unable to create block because world state is not available");
-                  return new CancellationException(
-                      "World state not available for block "
-                          + parentHeader.getNumber()
-                          + " with state root "
-                          + parentStateRoot);
-                });
-
-    return worldState.copy();
+    return protocolContext
+        .getWorldStateArchive()
+        .getMutable(parentStateRoot, parentHeader.getHash(), false)
+        .orElseThrow(
+            () -> {
+              LOG.info("Unable to create block because world state is not available");
+              return new CancellationException(
+                  "World state not available for block "
+                      + parentHeader.getNumber()
+                      + " with state root "
+                      + parentStateRoot);
+            })
+        // TODO: only copy if this is from layered worldstate
+        .copy();
```

Comment:
I need to re-review this after adding the snapshot TrieLogManager.  My original intent was to push the copy up into `getMutableSnapshot()`.  Right now copy() is safe but potentially unnecessary

Reply:
for the snapshot trie log manager, when we get a mutable non-persisting worldstate, it is giving a copy.  That way we can't have a rogue bit of code accidentally corrupting the snapshots we are keeping for the last 512 states.  

I am inclined to do this for the trie log manager used for the layered worldstate also and remove the copy() here.  
Then for read only uses like blockchain queries we can create a ReadOnlyWorldState so that we only bother to create copies when we are actually going to mutate the worldstate.  Thoughts?  I will probably wait and do this in a subsequent PR since this one is getting a bit unwieldy already.